test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/PackTaskLogicTests.cs

```diff
@@ -221,6 +221,69 @@ public void PackTaskLogic_SupportsContentTargetFolders()
         }
 
         [Theory]
+        [InlineData("", "abc.txt", "abc.txt")]
+        [InlineData("folderA", "abc.txt", "folderA/abc.txt")]
+        [InlineData("folderA;folderB", "abc.txt", "folderA/abc.txt;folderB/abc.txt")]
+        [InlineData("folderA;folderB\\subFolderA", "abc.txt", "folderA/abc.txt;folderB/subFolderA/abc.txt")]
+        [InlineData("folderA;folderB\\subFolderA;\\", "abc.txt", "folderA/abc.txt;folderB/subFolderA/abc.txt;abc.txt")]
+        public void PackTaskLogic_SupportsPackagePath_OnContent(string packagePath, string fileName, string expectedPackagePaths)
+        {
+            // Arrange
+            using (var testDir = TestDirectory.Create())
+            {
+                var tc = new TestContext(testDir);
+                var metadata = new Dictionary<string, string>()
+                {
+                    {"PackagePath", packagePath},
+                };
+                var msbuildItem = tc.AddContentToProject("", fileName, "hello world", metadata);
+                tc.Request.PackageFiles = new MSBuildItem[] { msbuildItem };
+                tc.Request.ContentTargetFolders = new string[] { "folderA", "folderB" };
+                // Act
+                tc.BuildPackage();
+
+                // Assert
+                Assert.True(File.Exists(tc.NuspecPath), "The intermediate .nuspec file is not in the expected place.");
+                Assert.True(File.Exists(tc.NupkgPath), "The output .nupkg file is not in the expected place.");
+                using (var nupkgReader = new PackageArchiveReader(tc.NupkgPath))
+                {
+                    var nuspecReader = nupkgReader.NuspecReader;
+
+                    // Validate the .nuspec.
```

Comment:
In general, a unit-test should verify a single aspect as declared. In this case location of the content files. Otherwise it's noisy and redundant.